extensions/content-manager/admin/src/containers/EditView/index.js

```diff
@@ -0,0 +1,317 @@
+    try {
+      const structureCorrectness = await request(`/content-manager/${collectionName}/validateRelation/${relationName}/${parentId}/${childId}`, {
+        method: 'GET',
+      });
+      setIsStructureCorrect(structureCorrectness.isCorrect)
+    } catch (err) {
+      strapi.notification.error(`content-manager.error.relation.fetch`);
+    }
+  };
+
+  useEffect(() => {
+    if(isRelationViewEnabled !== undefined &&  !_.isEmpty(selectedOption)){
+      getStructureCorrectness(collectionName, selectedRelationName, currentContentTypeId, selectedOption.id)
+    } else if(isRelationViewEnabled !== undefined) {
+      getStructureCorrectness(collectionName, selectedRelationName, currentContentTypeId, -1)
+    }
+  }, [selectedOption])
+
+  const formatLayoutRef = useRef();
+  formatLayoutRef.current = createAttributesLayout;
+  // Retrieve push to programmatically navigate between views
+  const { push } = useHistory();
+  // Retrieve the search and the pathname
+  const {
+    params: { contentType },
+  } = useRouteMatch('/plugins/content-manager/:contentType');
+  const isSingleType = contentType === 'singleType';
+  const [reducerState, dispatch] = useReducer(reducer, initialState, () => init(initialState));
+  const allLayoutData = useMemo(() => get(layouts, [slug], {}), [layouts, slug]);
+  const currentContentTypeLayoutData = useMemo(() => get(allLayoutData, ['contentType'], {}), [
+    allLayoutData,
+  ]);
+  const currentContentTypeLayout = useMemo(
+    () => get(currentContentTypeLayoutData, ['layouts', 'edit'], []),
+    [currentContentTypeLayoutData]
+  );
+  const currentContentTypeLayoutRelations = useMemo(
+    () => get(currentContentTypeLayoutData, ['layouts', 'editRelations'], []),
+    [currentContentTypeLayoutData]
+  );
+  const currentContentTypeSchema = useMemo(
+    () => get(currentContentTypeLayoutData, ['schema'], {}),
+    [currentContentTypeLayoutData]
+  );
+
+  const getFieldMetas = useCallback(
+    fieldName => {
+      return get(currentContentTypeLayoutData, ['metadatas', fieldName, 'edit'], {});
+    },
+    [currentContentTypeLayoutData]
+  );
+  const getField = useCallback(
+    fieldName => {
+      return get(currentContentTypeSchema, ['attributes', fieldName], {});
+    },
+    [currentContentTypeSchema]
+  );
+  const getFieldType = useCallback(
+    fieldName => {
+      return get(getField(fieldName), ['type'], '');
+    },
+    [getField]
+  );
+  const getFieldComponentUid = useCallback(
+    fieldName => {
+      return get(getField(fieldName), ['component'], '');
+    },
+    [getField]
+  );
+
+  // Check if a block is a dynamic zone
+  const isDynamicZone = useCallback(
+    block => {
+      return block.every(subBlock => {
+        return subBlock.every(obj => getFieldType(obj.name) === 'dynamiczone');
+      });
+    },
+    [getFieldType]
+  );
+
+  useEffect(() => {
+    // Force state to be cleared when navigation from one entry to another
+    dispatch({ type: 'RESET_PROPS' });
```

Comment:
use some constant instead of string - and maybe You can connect it instead of using dispatch and manually choosing action.type?

Reply:
strapi code